### ISSUE TITLE
/opt/homebrew/bin/yadm: line 660: local_classes: bad array subscript

### ISSUE THREAD

**Issue by davay** [TheLocehiliosan/yadm#411]
### Describe the bug

Anytime I run Yadm, though the command succeeds, it also throws an error message:

Note: Doesn't seem to be breaking anything..? 
```
opt/homebrew/bin/yadm: line 660: local_classes: bad array subscript
```

### To reproduce

1. Run command 'yadm status'
2. Run command 'yadm diff'
3. Run command 'yadm commit'

### Environment

 - Operating system: macOS 12.3 (21E230)
 - Version yadm: 3.2.0
 - Version Git: 2.35.1
 - Version bash: 3.2.57(1)-release

**Comment by TheLocehiliosan** [comment]
Can you also tell me what version of Bash is being used? (`yadm version` should report this)

**Comment by davay** [comment]
@TheLocehiliosan updated in the description, thanks

**Comment by TheLocehiliosan** [comment]
Thanks for the report. It looks like the implementation of multiple classes uses a Bash 4.2 feature. 

It shouldn't be too hard to fix.

I'm away from a computer right now, but I think line 660 needs to change

`local_class="${local_classes[-1]:-}"`

To

`local_class="${local_classes[${#local_classes[@]}-1]:-}"`

**Comment by TheLocehiliosan** [comment]
I only had a moment, but I pushed a possible fix to branch fix/411. If you can test that, and let me know how it goes.

Looking at the code, I also think it may be viable to just remove that line too.

I'll have to look tonight when I'm back.

**Comment by TheLocehiliosan** [comment]
I've been able to recreate this problem, and I've refined the fix a little.

I'll release 3.2.1 tonight, which addresses this problem.

**Comment by TheLocehiliosan** [comment]
Fixed in 3.2.1
